projects/core/components/dialog/dialog.component.ts

```diff
@@ -71,6 +77,12 @@ export class TuiDialogComponent<O, I> {
             : SMALL_DIALOGS_ANIMATION;
     }
 
+    @HostListener('click', ['$event.target']) hostClick(target: HTMLElement) {
+        if (target?.classList.contains('ng-trigger')) {
```

Comment:
@maartentibau if you want to check if target for click is the host component, a better way to do it is test `target === currentTarget`. cdk package has a utility function `isCurrentTarget` for this check.

Reply:
I was a bit confused with the comment, but after a search in the source code I found the `isCurrentTarget` method and understood what you mean... I first thought you were talking about the Angular Material CDK ... but that seemed a bit weird.